app/services/invitation/insert_from_invite.rb

```diff
@@ -69,6 +69,9 @@ def generate_password
     end
 
     def set_permission
+      if invite.user.network_partner?
```

Comment:
Maybe:

```invite.role = 'facilitator' if invite.user.network_partner?```

This is more of a style thing than anything else, and I'm fine with this as is.